components/sup/src/manager/service/config.rs

```diff
@@ -316,41 +316,39 @@ struct Cfg {
     environment: Option<toml::Value>,
 }
 
-// Shallow merges two toml tables.
-fn toml_merge(existing: &toml::Table, ovr: &toml::Table) -> toml::Table {
-    let mut final_map = existing.clone();
-    for (ovr_key, ovr_value) in ovr.iter() {
-
-        // attempt to merge nested tables
-        let existing_v = existing.get(ovr_key).and_then(|v| v.as_table());
-        let ovr_v = ovr.get(ovr_key).and_then(|v| v.as_table());
-        if let None = existing.get(ovr_key) {
-            final_map.insert(ovr_key.clone(), ovr_value.clone());
+// Recursively merges two TOML tables
+fn toml_merge(me: &toml::Table, other: &toml::Table) -> toml::Table {
+    let mut result = me.clone();
```

Comment:
Could we take a mutable reference to `me` instead of cloning the data? This is going to be quite costly